add community catalog proposal (#1)
percona/dbaas-catalog: docs/proposals/00-dbaas-community-catalog.md

    - [Monitoring Requirements](#monitoring-requirements)
    - [Dependencies](#dependencies)
    - [Troubleshooting](#troubleshooting)
  - [Implementation History](#implementation-history)
  - [Drawbacks](#drawbacks)
  - [Alternatives](#alternatives)
<!-- /toc -->

## Summary

<!--
This section is incredibly important for producing high-quality, user-focused
documentation such as release notes or a development roadmap. It should be
possible to collect this information before implementation begins, in order to
avoid requiring implementors to split their attention between writing release
notes and implementing the feature itself. KEP editors and SIG Docs
should help to ensure that the tone and content of the `Summary` section is
useful for a wide audience.

A good summary is probably at least a paragraph in length.

Both in this section and below, follow the guidelines of the [documentation
style guide]. In particular, wrap lines to a reasonable length, to make it
easier for reviewers to cite specific portions, and to minimize diff churn on
updates.

[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
-->

> OLM Catalogs are part of Operator Lifecycle Manager ([OLM](https://olm.operatorframework.io/)) framework to provide set of operators that extend Kubernetes. DBaaS Catalog would provide set of operators to extend Kubernetes to deliver DBaaS Platform based on it.

Reviewer: ```suggestion
OLM Catalogs are part of the Operator Lifecycle Manager ([OLM](https://olm.operatorframework.io/)) framework to provide a set of operators that extend Kubernetes. DBaaS Catalog would provide a set of operators to extend Kubernetes to deliver a DBaaS Platform based on these.
```

## Motivation

<!--
This section is for explicitly listing the motivation, goals, and non-goals of
this KEP.  Describe why the change is important and the benefits to users. The
motivation section can optionally provide links to [experience reports] to
demonstrate the interest in a KEP within the wider Kubernetes community.

[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
-->

> The Operator Lifecycle Manager ([OLM](https://olm.operatorframework.io/)) extends Kubernetes to provide a declarative way to install, manage, and upgrade Operators on a cluster. [OLM Catalogs](https://olm.operatorframework.io/docs/concepts/crds/catalogsource/) is the way to distribute set of operator that can be installed onto a cluster via OLM.

Reviewer: ```suggestion
The Operator Lifecycle Manager ([OLM](https://olm.operatorframework.io/)) extends Kubernetes to provide a declarative way to install, manage, and upgrade Operators on a cluster. [OLM Catalogs](https://olm.operatorframework.io/docs/concepts/crds/catalogsource/) are the way to distribute a set of operators that can be installed onto a cluster via OLM.
```

There are some catalogs already available:
  - [OperatorHub.io](https://operatorhub.io): https://github.com/k8s-operatorhub/community-operators
  - [RH Catalogs](https://docs.openshift.com/container-platform/4.6/operators/understanding/olm-rh-catalogs.html#olm-rh-catalogs_olm-rh-catalogs): https://github.com/redhat-openshift-ecosystem/community-operators-prod

> Existing catalogs have all possible operators and could be overkill for most of the systems, as well as for developers to ensure that some sub-set of it works good together to solve particular problems.

Reviewer: ```suggestion
Existing catalogs include all available operators. This could be overkill for most of the systems and for developers to ensure that some sub-set of it works well together to solve particular problems.
```

Links:
  - https://olm.operatorframework.io/
  - https://k8s-operatorhub.github.io/community-operators/
  - https://github.com/operator-framework/cool-catalog/

### Goals

<!--
List the specific goals of the KEP. What is it trying to achieve? How will we
know that this has succeeded?
-->

> Create and Maintain DBaaS catalog to provide instrumentation (catalog of operators) to build DBaaS Platform on Kubernetes base. Enable community to develop and extend this catalog.

Reviewer: ```suggestion
Create and maintain a DBaaS catalog to provide instrumentation (catalog of operators) to build a DBaaS Platform on Kubernetes. Enable the community to maintain and extend this catalog.
```

### Non-Goals

<!--
What is out of scope for this KEP? Listing non-goals helps to focus discussion
and make progress.
-->

Developing operators or DBaaS Platform.

## Proposal

<!--
This is where we get down to the specifics of what the proposal actually is.
This should have enough detail that reviewers can understand exactly what
you're proposing, but should not include things like API designs or
implementation. What is the desired outcome and how do we measure success?.
The "Design Details" section below is for the real
nitty-gritty.
-->

> Proposal is to create a new catalog of operators that solves one particular problem: DBaaS Platform. The scope of that catalog would be everything related to ability to build DBaaS Platform which includes but not limits to such [categories](https://github.com/k8s-operatorhub/community-operators/blob/main/categories.json):

Reviewer: ```suggestion
The proposal is to create a new catalog of operators that solves one particular problem: setting up a DBaaS Platform. The scope of that catalog would be everything related to the ability to build DBaaS Platform, which includes but is not limited to the following [categories](https://github.com/k8s-operatorhub/community-operators/blob/main/categories.json):
```

  - Cloud Provider
  - Database
  - Logging & Tracing
  - Monitoring

> Such catalog will allow to separate different concerns, to have smaller repos/images/systems, to harden security by reducing the scope.

Reviewer: ```suggestion
Such a catalog will allow for separating different concerns, having smaller repos/images/systems, for hardening security by reducing the scope.
```

### User Stories (Optional)

<!--
Detail the things that people will be able to do if this KEP is implemented.
Include as much detail as possible so that people can understand the "how" of
the system. The goal here is to make this feel real for users without getting
bogged down.
-->

#### Story 1

> As the SRE I would like to have a Kubernetes cluster dedicated for DB deployments and install all needed operators for only that specific use case with OLM.

Reviewer: ```suggestion
As an SRE, I would like to set up a Kubernetes cluster dedicated to database deployments and install all required operators for only that specific use case with OLM.
```

### Notes/Constraints/Caveats

<!--
What are the caveats to the proposal?
What are some important details that didn't come across above?
Go in to as much detail as necessary here.
This might be a good place to talk about core concepts and how they relate.
-->

#### DBaaS Platform

> To set the base for motivation lets imagine what DBaaS Platform could be (so there would imaginary proposal for the imaginary problem).

Reviewer: ```suggestion
To set the basis for motivation, let's imagine what a DBaaS Platform could look like (so there would be an imaginary proposal for the imaginary problem).
```

DBaaS Platform is a platform that extends Kubernetes with ability to:
  - deploy Databases and Database clusters (in any infrastructure)
  - provision infrastructure for Database \[Cluster\]
  - Observability for Databases and infrastructure
  - Management and Operations for Databases

### Risks and Mitigations

<!--
What are the risks of this proposal, and how do we mitigate? Think broadly.
For example, consider both security and how this will impact the larger
Kubernetes ecosystem.

How will security be reviewed, and by whom?

How will UX be reviewed, and by whom?

Consider including folks who also work outside the SIG or subproject.
-->

Drifts from upstreams (other catalogs, operator versions).

Version Skew between Catalogs.

## Design Details

<!--
This section should contain enough information that the specifics of your
change are understandable. This may include API specs (though not always
required) or even code snippets. If there's any ambiguity about HOW your
proposal will be implemented, this is the place to discuss them.
-->

Another proposal would be submitted for the DBaaS catalog design.

> For community based catalog there should be a design that enables community and community contributions.

Reviewer: ```suggestion
For a community-based catalog, a design should enable community and community contributions.
```

### Dependencies

<!--
This section must be completed when targeting beta to a release.
-->

### Troubleshooting

<!--
This section must be completed when targeting beta to a release.

For GA, this section is required: approvers should be able to confirm the
previous answers based on experience in the field.

The Troubleshooting section currently serves the `Playbook` role. We may consider
splitting it into a dedicated `Playbook` document (potentially with some monitoring
details). For now, we leave it here.
-->

## Implementation History

<!--
Major milestones in the lifecycle of a KEP should be tracked in this section.
Major milestones might include:
- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
- the `Proposal` section being merged, signaling agreement on a proposed design
- the date implementation started
- the first Kubernetes release where an initial version of the KEP was available
- the version of Kubernetes where the KEP graduated to general availability
- when the KEP was retired or superseded
-->

## Drawbacks

<!--
Why should this KEP _not_ be implemented?
-->

> Global catalog could be enough as it takes not that much space and maybe lucks better API on categories to select needed operators for installation.

Reviewer: ```suggestion
A global catalog might be sufficient as it does not require that much space and maybe only needs a better API to select the desired operators based on categories.
```
